seaworthy/logs.py

```diff
@@ -12,6 +12,25 @@ def _last_few_log_lines(container, max_lines=100):
     return '\nLast few log lines:\n{}'.format(logs)
 
 
+def stream_with_history(container, timeout=10, **logs_kwargs):
+    """
+    Return an iterator over all container logs, past and future.
+
+    The docker API we use in stream_logs() won't give us old logs, and the
+    docker API container.logs() uses seems to sometimes block indefinitely on
+    the last few lines. To get around this, we fetch all the historical logs
+    before we start streaming the new logs.
+    """
+    # Start streaming immediately after fetching the old logs to minimise the
+    # chance of a race condition.
+    old_logs = container.logs(**logs_kwargs)
```

Comment:
Do we want to make sure `stream` is false?

Reply:
I was on the fence about that, but your question indicates that we should.

Reply:
Or rather, I was on the fence about rejecting the stream parameter. It needs to be `True` for `stream_logs()`, but `False` for `container.logs()`.